procedures/MT/rotator/rotator_performance.ipynb

```diff
@@ -0,0 +1,655 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #10.        data = csc.evt_summaryState.get()
</pre><p>shouldn't you use <code>aget</code> to get the last state (include a timeout incase one hasn't been published).</p><p> </p><p>Then the sleeps should not be necessary. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-ts/ts_notebooks/pull/96/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
I thought @r-owen  mentioned before that he recommended to use the `get()` instead of `aget()`.
In addition, for my personal experience, it is better to use the `get()` if the summit internet is not very stable.
Thanks!

Reply:
To expand on this: if you know that a topic has a value then you should use `get` to get the current value, or `next` if you want to wait for a value. `aget` is only intended to be used when code starts up and you are not sure if the topic has ever read any data at all. I regret adding it and discourage its use -- especially calling it more than once for any topic.

The big problem is that the behavior if `aget` can be surprising if there is no current value: it waits for a new value but does NOT consume that value from the queue. This means a subsequent call to `next(flush=False)` will return the same exact same data. There are good reasons to do that (e.g. `get` and `aget` never interfere with `next`). But there are also good reasons to do it the other way (e.g. one obvious, but incorrect, mental model of `aget` is to call `get`, and if the result is None then call `next(flush=False)`). The result of all this is that code using `aget` is ambiguous: it is easy to assume the wrong behavior.

Reply:
Thanks @r-owen . It'd be great if you could add that explanation to the salobj docs. 

Or can you start the deprecation process for `aget`? Presumably it'll take awhile... but if it throws a warning then it'll at least discourage it's use in the future.